Application.js

```diff
@@ -113,7 +129,7 @@ define(["require", "dcl/dcl", "decor/Stateful", "decor/Evented", "dojo/Deferred"
 				} else {
 					this.constraint = "center";
 				}
-				when(controllers, function () {
+				return Promise.resolve(controllers).then(function () {
```

Comment:
Yes, since I changed the functions to always return a promise the Promise.resolve is not needed. Thanks.